imagedephi/rules/ruleset.py

```diff
@@ -1,49 +1,88 @@
-        return []
 
+class SvsFormatRules(TiffFormatRules):
+    file_format = FileFormat.SVS
+    image_description_rules: dict[str, MetadataSvsRule]
 
-def _build_rule(file_format: FileFormat, rule_spec: dict, rule_source: RuleSource) -> Rule:
-    rule_type = RuleType[rule_spec["type"].upper()]
-    if file_format == FileFormat.TIFF:
-        if rule_type == RuleType.METADATA:
-            return MetadataTiffRule.build(rule_spec, rule_source)
-    if file_format == FileFormat.SVS:
-        if rule_type == RuleType.METADATA:
-            return MetadataSvsRule.build(rule_spec, rule_source)
+    @staticmethod
+    def build_svs_image_description_rule(
+        key_name: str, method_spec: str | dict, rule_source: RuleSource
+    ) -> MetadataSvsRule:
+        rule_spec = {"key_name": key_name}
+        if isinstance(method_spec, str):
+            rule_spec["method"] = method_spec
+        else:
+            rule_spec = rule_spec | method_spec
+        return MetadataSvsRule.build(rule_spec, rule_source)
+
+    def __init__(self, file_format_spec: dict, rule_source: RuleSource):
+        super().__init__(file_format_spec, rule_source)
+        self.image_description_rules = {}
+        for key_name, method_info in file_format_spec.get("image_description", {}).items():
+            self.image_description_rules[
+                key_name
+            ] = SvsFormatRules.build_svs_image_description_rule(key_name, method_info, rule_source)
+
+
+@dataclass
+class RuleSet:
+    name: str
+    description: str
+    tiff: TiffFormatRules
+    svs: SvsFormatRules
 
-    raise Exception("Unsupported rule.")
+    def get_format_rules(self, file_format: FileFormat) -> FileFormatRules:
+        if file_format == FileFormat.TIFF:
+            return self.tiff
+        if file_format == FileFormat.SVS:
+            return self.svs
+        raise Exception(f"File format {file_format} not supported")
 
 
 def build_ruleset(ruleset_spec: dict, rule_source: RuleSource) -> RuleSet:
     """Read in metadata redaction rules from a file."""
-    rule_set_rules = {}
-    for file_format_key, rule_specs in ruleset_spec["rules"].items():
-        file_format = FileFormat[file_format_key.upper()]
-        rule_set_rules[file_format] = [
-            _build_rule(file_format, rule_spec, rule_source) for rule_spec in rule_specs
-        ]
-    return RuleSet(ruleset_spec["name"], ruleset_spec["description"], rule_set_rules)
+    rule_set_args = [ruleset_spec["name"], ruleset_spec["description"]]
+    for file_format in FileFormat:
+        format_rules = ruleset_spec.get(file_format.value, None)
+        if not format_rules:
+            raise Exception(f"No rules provided for format {file_format}")
+        if file_format == FileFormat.TIFF:
+            rule_set_args.append(TiffFormatRules(format_rules, rule_source))
+        elif file_format == FileFormat.SVS:
+            rule_set_args.append(SvsFormatRules(format_rules, rule_source))
+    return RuleSet(*rule_set_args)
```

Comment:
I should rework this to avoid having to pass an args list to the `RuleSet` constructor. Mypy was giving me a tough time because the tiff/svs rules may not be defined by the time we hit the return statement.